clang/lib/StaticAnalyzer/Checkers/StreamChecker.cpp

```diff
@@ -342,6 +342,11 @@ class StreamChecker : public Checker<check::PreCall, eval::Call,
                          CheckerContext &C,
                          const StreamErrorState &ErrorKind) const;
 
+  /// Perform some common checks for all preXXX functions.
+  bool basicCheck(const FnDescription *Desc, const CallEvent &Call,
+                  CheckerContext &C, ProgramStateRef &State,
+                  SVal &StreamVal) const;
```

Comment:
This function has a role like `ensureStreamNonNullAndOpened`, that would be a better name for it. And it should return the  new `State`, like `ensureStreamNonNull`.